OJS.Workers.ExecutionStrategies/RunSpaAndExecuteMochaTestsExecutionStrategy.cs

```diff
@@ -0,0 +1,343 @@
+﻿namespace OJS.Workers.ExecutionStrategies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Ionic.Zip;
+    using OJS.Workers.Common;
+    using OJS.Workers.Common.Extensions;
+    using OJS.Workers.Common.Helpers;
+    using OJS.Workers.Common.Models;
+    using OJS.Workers.ExecutionStrategies.Models;
+    using OJS.Workers.ExecutionStrategies.Python;
+    using OJS.Workers.Executors;
+
+    public class RunSpaAndExecuteMochaTestsExecutionStrategy: PythonExecuteAndCheckExecutionStrategy
+    {
+        protected const string UserApplicationPathPlaceholder = "#userApplicationPath#";
+        protected const string UserApplicationHttpPortPlaceholder = "#userApplicationHttpPort#";
+        protected const string TestsPathPlaceholder = "#testsPath#";
+        protected const string NodeModulesRequirePattern = "(require\\(\\')([\\w]*)(\\'\\))";
+        protected const string MochaTestsPassingFailingResultPattern = "([\\d]*)\\s*(passing|failing)";
+
+        public RunSpaAndExecuteMochaTestsExecutionStrategy(
+            IProcessExecutorFactory processExecutorFactory,
+            string pythonExecutablePath,
+            string jsProjNodeModulesPath,
+            string mochaNodeModulePath,
+            string chaiNodeModulePath,
+            string playwrightModulePath,
+            int portNumber,
+            int baseTimeUsed,
+            int baseMemoryUsed)
+            : base(
+                  processExecutorFactory,
+                  pythonExecutablePath,
+                  baseTimeUsed,
+                  baseMemoryUsed)
+        {
+            this.JSProjNodeModulesPath = jsProjNodeModulesPath;
+            this.MochaModulePath = mochaNodeModulePath;
+            this.ChaiModulePath = chaiNodeModulePath;
+            this.PlaywrightModulePath = playwrightModulePath;
+            this.PortNumber = portNumber;
+        }
+
+        protected int PortNumber { get; }
+
+        protected Dictionary<string, string> NodeModulesPaths { get; }
+       
+        protected string MochaModulePath { get; }
+
+        protected string ChaiModulePath { get; }
+
+        protected string PlaywrightModulePath { get; }
+
+        protected string JSProjNodeModulesPath { get; }
+
+        protected string TestsPath => FileHelpers.BuildPath(this.WorkingDirectory, "test");
+
+        protected string UserApplicationPath => FileHelpers.BuildPath(this.WorkingDirectory, "app");
+
+        protected string NgingConfFilePath => FileHelpers.BuildPath(this.WorkingDirectory, "nginx.conf");
```

Comment:
strings `test`, `app` and `nginx.conf` can be extracted into local constants, i..e constants in this file